inputremapper/injection/macros/macro.py

```diff
@@ -482,9 +483,25 @@ def add_mouse(self, direction: str, speed: int):
         }[direction.lower()]
 
         async def task(handler: Callable):
-            resolved_speed = value * _resolve(speed, [int])
+            resolved_speed = _resolve(speed, [int])
+            resolved_accel = _resolve(acceleration, [float, None])
+
+            if resolved_accel:
+                current_speed = 0.0
+                displacement_accumulator = 0.0
+                displacement = 0
+            else:
+                displacement = resolved_speed
+
             while self.is_holding():
-                handler(EV_REL, code, resolved_speed)
+                if resolved_accel and current_speed < resolved_speed:
+                    current_speed += resolved_accel
+                    current_speed = min(current_speed, resolved_speed)
+                    displacement_accumulator += current_speed
+                    displacement = int(displacement_accumulator)
+                    displacement_accumulator -= displacement
```

Comment:
Sorry, one more thing: I'd like to have an explanatory comment there why we are doing this, to avoid any future confusion.

```suggestion
                # Cursors can only move by integers. To get smooth acceleration for
                # small acceleration values, the cursor needs to move by a pixel every
                # few iterations. This can be achieved by remembering the decimal
                # places that were cast away, and using them for the next iteration.
                if resolved_accel and current_speed < resolved_speed:
                    current_speed += resolved_accel
                    current_speed = min(current_speed, resolved_speed)
                    displacement_accumulator += current_speed
                    displacement = int(displacement_accumulator)
                    displacement_accumulator -= displacement
```